src/web3-wrapper/rpc/rpcOracle.ts

```diff
@@ -1,34 +1,37 @@
-import {CHAINID} from '../../constants';
+import { CHAINID } from '../../constants';
+import { RpcInfo } from '../../web3-wrapper/rpc/rpcInfo';
 
 export class RPCOracle {
-  private rpcs: string[];
-  private currentIndex = -1; // Initialize currentIndex
+  private rpcInfos: RpcInfo[];
 
-  constructor(private networkId: number | string, rpcUrls: string[]) {
+  constructor(
+    private networkId: number | string,
+    rpcInfos: RpcInfo[],
+  ) {
     // check if networkId passed as parameter exists in CHAINID Enum
     if (!Object.entries(CHAINID).some((e) => e[1] === String(networkId))) {
       throw new Error(`Chain with ID ${networkId} not found.`);
     }
-    this.rpcs = rpcUrls;
+    this.rpcInfos = rpcInfos;
   }
 
   getRpcCount(): number {
-    return this.rpcs.length;
+    return this.rpcInfos?.length || 0;
   }
 
-  getNextAvailableRpc() {
-    if (this.rpcs.length === 0) {
-      return null;
+  getNextAvailableRpc(): string {
+    let totalWeight = 0;
+    for (const rpc of this.rpcInfos) {
+      totalWeight += rpc.weight;
     }
```

Comment:
don't use for loop here, use .reduce() https://medium.com/@onlinemsr/top-6-easy-ways-to-sum-an-array-in-javascript-41491c305fd2